src/spoke/Spoke.sol

```diff
@@ -87,12 +87,14 @@ contract Spoke is Auth, Recoverable, ReentrancyProtection, ISpoke, ISpokeGateway
     //----------------------------------------------------------------------------------------------
 
     /// @inheritdoc ISpoke
-    function transferShares(uint16 centrifugeId, PoolId poolId, ShareClassId scId, bytes32 receiver, uint128 amount)
-        external
-        payable
-        payTransaction
-        protected
-    {
+    function crossTransferShares(
```

Comment:
Nit, I would do
```suggestion
    function crosschainTransfer(
```

Matches similar naming in https://eips.ethereum.org/EIPS/eip-7802

Reply:
I would leave the "Shares" name, it's true that the EIP-7802 doesn't say shares, but it is in the context of shares. From our level can be confused with assets

Reply:
Yeah fair, then would do `crosschainTransferShares`